load_testing/common_flows/flow_sp_sign_in.py

```diff
@@ -98,3 +140,245 @@ def do_sp_sign_in(context):
+            "authenticity_token": auth_token,
+        }
+    )
+    resp = do_request(context, "get", "/", "/")
+    auth_token = authenticity_token(resp)
+
+    # Post login credentials
+    resp = do_request(
+        context,
+        "post",
+        "/",
+        "/",
+        {
+            "user[email]": credentials["email"],
+            "user[password]": "bland pickles",
+            "authenticity_token": auth_token,
+        },
+    )
+
+    # Validate that we got the expected response and were not redirect back for
+    # some other reason.
+    if resp.text.find('The email or password you’ve entered is wrong') == -1:
+        print(f"ERROR: the expected response for incorrect l/p is not present")
+
+    return resp
+
+
+def do_sign_in_incorrect_sms_otp(context):
+    sp_root_url = get_env("SP_HOST")
+    context.client.cookies.clear()
+
+    # GET the SP root, which should contain a login link, give it a friendly
+    # name for output
+    resp = do_request(
+        context,
+        "get",
+        sp_root_url,
+        sp_root_url,
+        {},
+        {},
+        sp_root_url
+    )
+
+    sp_signin_endpoint = sp_root_url + '/auth/request?aal=&ial=1'
+    # submit signin form
+    resp = do_request(
+        context,
+        "get",
+        sp_signin_endpoint,
+        '',
+        {},
+        {},
+        sp_signin_endpoint
+    )
+    auth_token = authenticity_token(resp)
+    request_id = querystring_value(resp.url, "request_id")
+
+    # This should match the number of users that were created for the DB with
+    # the rake task
+    num_users = get_env("NUM_USERS")
+    credentials = random_cred(num_users)
+
+    # POST username and password
+    resp = do_request(
+        context,
+        "post",
+        "/",
+        "/",
+        {
+            "user[email]": credentials["email"],
+            "user[password]": credentials["password"],
+            "user[request_id]": request_id,
+            "authenticity_token": auth_token,
+        }
+    )
+    resp = do_request(context, "get", "/", "/")
+    auth_token = authenticity_token(resp)
+
+    # Post login credentials
+    resp = do_request(
+        context,
+        "post",
+        "/",
+        "/login/two_factor/sms",
+        {
+            "user[email]": credentials["email"],
+            "user[password]": credentials["password"],
+            "authenticity_token": auth_token,
+        },
+    )
+    auth_token = authenticity_token(resp)
+
+    # Post to unauthenticated redirect
+    resp = do_request(
+        context,
+        "post",
+        "/login/two_factor/sms",
+        "/login/two_factor/sms",
+        {"code": "000000", "authenticity_token": auth_token},
+    )
+
+    # Validate that we got the expected response and were not redirect back for
+    # some other reason.
+    if resp.text.find('That security code is invalid.') == -1:
+        print(f"ERROR: the expected response for incorrect OTP is not present")
```

Comment:
No interpolation in the string, so you can remove the 'f' before "

Reply:
fixed